catalog_tools/analysis/estimate_beta.py

```diff
@@ -230,6 +233,167 @@ def estimate_b_laplace(
+        magnitudes: vector of earthquake magnitudes
+        years: vector of years of occurrence of each earthquake
+        completeness_table: Nx2 array, where the first column
+            contains the leftmost edge of magnitude bins and
+            the second column the associated year of completeness, i.e.
+            the year after which all earthquakes larger than the value in
+            the first column are considered detected. An example is given
+            below:
+
+            np.array([[ 3.95, 1980],
+                      [ 4.95, 1920],
+                      [ 5.95, 1810],
+                      [ 6.95, 1520]])
+
+        mag_max: maximum possible magnitude
+        tend: last year of observation (the default is None, in which case
+              it is set to the latest year in years).
+        delta_m: magnitude resolution, the default is 0.1.
+        b_parameter:either 'b-value', then the corresponding value of the
+                    Gutenberg-Richter law is returned, otherwise 'beta'
+                    from the exponential distribution [p(M) = exp(-beta*(M-mc))]
+
+    Returns:(
+        b_parameter: maximum likelihood point estimate of 'b-value' or 'beta'
+        std_b_parameter: standard error of b_parameter
+        rate_at_lmc: maximum likelihood point estimate of earthquake rate
+                     at the lower magnitude of completeness
+        std_rate_at_lmc: standard error of rate_at_lmc
+        a_val: maximum likelihood point estimate of a-value
+               ( =log10(rate at mag=0) ) of Gutenberg-Richter
+               magnitude frequency distribution
+    """
+    assert len(magnitudes) == len(years), \
+        "the magnitudes and years arrays have different lengths"
+    assert completeness_table.shape[1] == 2
+    assert np.all(np.ediff1d(completeness_table[:, 0]) >= 0),\
+        "magnitudes in completeness table not in ascending order"
+    assert [i - delta_m in np.arange(completeness_table[0, 0],
+                                     mag_max + 0.001, delta_m)
+            for i in np.unique(magnitudes)],\
+        "magnitude bins not aligned with completeness edges"
+    if not np.all(magnitudes >= completeness_table[:, 0].min()):
+        warnings.warn(
+            "magnitudes below %.2f are not covered by the "
+            "completeness table and are discarded" %
+            completeness_table[0, 0])
+    assert b_parameter == 'b_value' or b_parameter == 'beta', \
+        "please choose either 'b_value' or 'beta' as b_parameter"
+    factor = 1 / np.log(10) if b_parameter == 'b_value' else 1
+    tend = tend if tend else np.max(years)
+
+    # Get the magnitudes and completeness years as separate arrays
+    completeness_magnitudes = completeness_table[:, 0]
+    completeness_years = completeness_table[:, 1]
+
+    # Obtain the completeness start year for each value in magnitudes
+    insertion_indices = np.searchsorted(completeness_magnitudes, magnitudes)
+    cmpl_yrs = np.array(
```

Comment:
cmpl_yrs: not clear what this is from variable name, would it make sense to also call this completeness_years?

Reply:
Changed it to 'completeness_starts'. Also changed what was called completeness_magnitudes, completeness_years to completeness_table_magnitudes, completeness_table_years